plugins/fileservices/fileservices.cpp

```diff
@@ -289,25 +290,30 @@ static const char *getAccessibleEspServerURL(const char *param, IConstWorkUnit *
                 ForEach(*espBindingIter)
                 {
                     espBindingIter->query().getProp("@service",wsFSUrl.clear());
-                    if (wsFSUrl.length() && (stricmp(wsFSUrl.str(),"EclWatch")==0 || stricmp(wsFSUrl.str(),"ws_fs")==0))
+                    xpath.setf("Software/EspService[@name=\"%s\"]/Properties/@type", wsFSUrl.str());
+
+                    if(env->getProp(xpath.str(), espServiceType.clear()))
                     {
-                        if (espBindingIter->query().getProp("@protocol",bindingProtocol.clear()))
+                        if (espServiceType.length() && (stricmp(espServiceType.str(),"WsSMC")==0 || stricmp(espServiceType.str(),"FileSpray_Serv")==0))
```

Comment:
Minors: use isEmpty() to replace length() and use strieq() to replace stricmp().